### ISSUE TITLE
Have an MSI installer

### ISSUE THREAD

**Issue by fesch** [fesch/Structorizer.Desktop#871]
Someone mailed to ask if we could provide an MSI installer, which would allow an easy deployment in larger Windows environments.

Searching the net, I found the WiX Toolset, which I will try to get into ...

https://www.firegiant.com/wix/tutorial/

**Comment by fesch** [comment]
I've succeeded in using the WiX Toolset to create an MSI package that makes a silent install of Structorizer onto the disk.

I've not yet found the right switches to install in user mode, so Upla updates are disabled by default. I think this is tine for a deployment in a large Windows network, especially as Structorizer is not big in size.

All others should use the Inno generated EXE installer, which offer a GUI with all the features ;)

**Comment by codemanyak** [comment]
> I've not yet found the right switches to install in user mode, so Upla updates are disabled by default.

So, the administrator would have to reinstall Structorizer each time a new version is published?
> All others should use the Inno generated EXE installer, which offers a GUI with all the features ;)

Well, but to offer two different Windows installers with different update service might confuse users as they may usually tend to prefer the msi package. Requires a lot of documentation.
And what about installation of both? Would they interfere? I hoped that the msi package might be functionally compatibe with the Inno installer and would ideally accept the latter as an equivalent product.

**Comment by fesch** [comment]
Yes, but such deployments are made server-side. Kind of a one-click action ...

No, the installers do not detect each other, even tough I've used the same application ID and GUID :-( As far as I've read in the documentation, WiX should be able to do more or less the same thing Inno can, but I need to learn how to implement it.

For now, I've just got a POC that WiX is working, so I continue to dig into it ...

**Comment by fesch** [comment]
After having read the entire weekend about how to create an MSI installer using the WIX Toolkit, I finally succeeded. Here come the first draft.

In fact, I copied the strings from the Inno installer and tried to mimic is behaviour.

[Structorizer.zip](https://github.com/fesch/Structorizer.Desktop/files/4742520/Structorizer.zip)

**Comment by fesch** [comment]
Small fixes ...

[Structorizer.zip](https://github.com/fesch/Structorizer.Desktop/files/4742660/Structorizer.zip)

P.S.: Yes, I know that there is a upla[0..2].ini file that is not needed inside the installation directory, but I did not yet get the right command to delete it :-(

**Comment by codemanyak** [comment]
Sorry. Your second msi result aborts with a error on Windows 10, which is also reproducible on Vista:
![Error_msi_Windows10](https://user-images.githubusercontent.com/15326471/83979219-fd254000-a90c-11ea-9551-2f7eaf679e1b.png)
![Error_msi_Vista](https://user-images.githubusercontent.com/15326471/83979229-07dfd500-a90d-11ea-8ea7-87ec3c08ef38.png)

I don't blame you. I had also tried to work my way through the WiX tutorial: It seems powerful, but the explanations about e.g. installation directories and conditions, and in particular the configuration syntax weren't so obvious to understand, that I had to give it up for now in favour of more urgent tasks.

**Comment by fesch** [comment]
Ok, I'll try to debug tomorrow on some other machine than mine ;)

**Comment by LittleITworld** [comment]
any updates or ways to install structorizer silent?

**Comment by codemanyak** [comment]
@LittleITworld 
Um, no. I am afraid none of us has found the time to push this any further. Maybe both of us thought the other one might occupy with it, maybe it didn't seem that important. Sorry for having neglected this request in favour of the many others.
